File: Algorithms/Dynamic_Programming/Fibonacci_Series/fibonacci.js
/*
This function returns the nth fibonacci number, given 
*/
function fibonacci(n, memoizer) {
    memoizer = memoizer || {};
    if (memoizer[n]) {
        return memoizer[n];
    }
    if (n <= 1) {
        return 1;
    }
    return memoizer[n] = fibonacci(n - 1, memoizer) + fibonacci(n - 2, memoizer);
}

File: Algorithms/Dynamic_Programming/Fibonacci_Series/Fibonacci.cs
using System;

namespace Fibonacci
{
    public class Program
    {
       public static void Main(string[] args)
        {
            int n;

            Console.Write("Enter the number of terms: ");
            n  = Convert.ToInt32(Console.ReadLine());
            fibonacci(n);

        }

        public static void fibonacci(int n)
        {
            long a = 0;
            long b = 1;

            for (int i = 0; i < n; i++)
            {
                Console.WriteLine(b);
                b += a;
                a = b - a;
            }
        }
    }
}
